lib/rules/no-extra-parens.js

```diff
@@ -395,13 +413,14 @@ module.exports = {
          */
         function dryBinaryLogical(node) {
             const prec = precedence(node);
-            const shouldSkipLeft = NESTED_BINARY && (node.left.type === "BinaryExpression" || node.left.type === "LogicalExpression");
+            const shouldSkipLeft = (NESTED_BINARY && (node.left.type === "BinaryExpression" || node.left.type === "LogicalExpression")) ||
+              node.left.type === "UnaryExpression" && node.operator === "**";
```

Comment:
Before updating the `precedence` function, the rule incorrectly considered this code valid:

```js
async function foo(){
  (await foo) ** bar;
}
```